ui: metrics time selector unit test coverage

vladlos opened the pull request:
converted `<TimeScaleDropdown />` to functional component due to problems with `withRouter` hoc in test env
added unit tests for `<TimeScaleDropdown />` and `<TimeFrameControls />`

Release note (ui): none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46259)
<!-- Reviewable:end -->

koorosh reviewed (changes requested):
Got an error on project build:
```
ERROR: cockroachdb/cockroach/pkg/ui/src/views/cluster/components/controls/controls.spec.tsx:41:4 - file should end with a newline
```
Please take a look at this error.

koorosh commented on pkg/ui/src/views/cluster/components/controls/controls.spec.tsx:
```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 The Cockroach Authors.
```
Please, update year. 2018 -> 2020

koorosh reviewed (commented):
@elkmaster , I have a concern about `TimeFrameControls` component and props it exposes.
- `disabledArrows` prop makes you iterate three times over the same array to check if buttons have to be disabled.
-  Would it be more clear to name props as `onButtonClick` instead of `onArrowClick`
- it is only three buttons in the component and I suppose it can be more meaningful to change  `disabledArrows` prop to three props like: `isBackButtonDisabled`, `isForwardButtonDisabled`, and `isNowButtonDisabled` props.

elkmaster commented on pkg/ui/src/views/cluster/containers/timescale/index.tsx:
```diff
@@ -62,25 +69,93 @@ export const getTimeLabel = (currentWindow?: timewindow.TimeWindow, windowSize?:
-  arrowClick = (direction: ArrowDirection) => {
-    const { currentWindow, currentScale } = this.props;
-    const windowSize: any = moment.duration(moment(currentWindow.end).diff(currentWindow.start));
+  const arrowClick = (direction: ArrowDirection) => {
```
Arrow direction is not used as dropdown icon direction. Not great naming due to the fact as on previous designs it was arrows. But I would do not refactor that part in terms of this pr as this is about tests only. Let's revisit it in terms of integrating DS component here.

koorosh commented on pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx:
```diff
@@ -0,0 +1,175 @@
+// licenses/APL.txt.
+
+import React from "react";
+// import { assert } from "chai";
```
Remove commented imports

koorosh commented on pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx:
```diff
@@ -0,0 +1,175 @@
+  "#/metrics/overview/node/3?start=1584528726&end=1584529326", // Node 3 - 10 minutes
+];
+
+const dateFormat = "MMM DD,";
```
Would it be better to reuse `dateFormat` and `timeFormat` by exporting them and importing them in tests? Will reduce code duplication.

elkmaster commented on pkg/ui/src/views/cluster/containers/timescale/index.tsx:
```diff
@@ -62,25 +69,93 @@ export const getTimeLabel = (currentWindow?: timewindow.TimeWindow, windowSize?:
+    return {
+      dateStart: isSameStartDay
+        ? ""
+        : moment.utc(currentWindow.start).format(dateFormat),
```
good catch, added utc for L79-80

koorosh commented on pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx:
```diff
@@ -0,0 +1,175 @@
+  it("getTimeRangeTitle must return title Past 10 Minutes", () => {
+    const title = getTimeRangeTitle(state.currentWindow, state.currentScale);
+    const wrapper = makeTimeScaleDropdown(state);
+    console.log(wrapper.find(".trigger .Select-value-label"));
```
Remove `console.log`

elkmaster commented:
> @elkmaster , I have a concern about `TimeFrameControls` component and props it exposes.
> 
> * `disabledArrows` prop makes you iterate three times over the same array to check if buttons have to be disabled.
> * Would it be more clear to name props as `onButtonClick` instead of `onArrowClick`
> * it is only three buttons in the component and I suppose it can be more meaningful to change  `disabledArrows` prop to three props like: `isBackButtonDisabled`, `isForwardButtonDisabled`, and `isNowButtonDisabled` props.

I agree but would like to revisit this in terms of other ticket. This pr is about test coverage only.

koorosh commented on pkg/ui/src/views/cluster/containers/timescale/timescale.spec.tsx:
```diff
@@ -0,0 +1,175 @@
+import { MemoryRouter } from "react-router";
+import TimeFrameControls from "../../components/controls";
+import RangeSelect from "../../components/range";
+import { expect, assert } from "chai";
```
What is a reason to use `expect` for assertions? By default `assert` is used across the entire project.

koorosh reviewed (changes requested):
Tests look good to me, but code still needs to be cleaned up. Please take a look at my comments.

elkmaster commented on pkg/ui/src/views/cluster/components/controls/controls.spec.tsx:
```diff
@@ -0,0 +1,41 @@
+  const makeTimeScaleDropdown = (props: RangeSelectProps) => shallow((<TimeFrameControls {...props} />));
+
+  beforeEach(function() {
+    spy = sinon.spy();
```
that was original intent, but it is not worked well, so I switched to class based tests. Removed leftovers

vladlos committed:
ui: metrics time selector unit test coverage  …

converted <TimeScaleDropdown /> to functional component due to problems with `withRouter` hoc in test env
added unit tests for <TimeScaleDropdown /> and <TimeFrameControls />

Release note (ui): none

koorosh reviewed (approved):
LGTM

elkmaster commented:
bors r+

elkmaster commented:
bors r+

dhartunian commented:
bors r+

elkmaster commented:
bors r+
